File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> seen = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int x = nums[i];
            if(seen.containsKey(target - x)){
                return new int[]{seen.get(target - x), i};
            }
            seen.put(x, i);
        }
        return new int[]{};
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-2.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m = len(matrix)
        n = len(matrix[0])
        l = -1
        r = m * n
        while l + 1 < r:
            mid = (l + r) // 2
            x = matrix[mid // n][mid % n]
            if x == target:
                return True
            elif x < target:
                l = mid
            else:
                r = mid
        return False